doc/site/modules/core/list.markdown

```diff
@@ -94,6 +94,25 @@ System.print(["a", "b", "c"].removeAt(1)) //> b
 
 It is a runtime error if the index is not an integer or is out of bounds.
 
+### **sort**(), **sort**(comparer)
+
+Sorts the elements of a list in-place; altering the list.
+
+<pre class="snippet">
+var list = [4, 1, 3, 2].sort()
+System.print(list) //> [1, 2, 3, 4]
+</pre>
+
+A comparison function `comparer` can be provided to customise the element sorting. The comparison function must return a boolean value specifying the order in which elements should appear in the list. The default behaviour uses a `<` comparison.
```

Comment:
This doesn't explain the boolean result, or the nature of the compare function :) We'd ideally do both, I can adjust the documentation after merge though, just wanted to give feedback.

> The comparison function accepts two arguments, which are the two values to compare, and must return a boolean value to indicate if the first value is smaller than the second value. If the function returns true, the first argument will be before the second in the sorted results. 
> 
> A compare function like `{|value, other| true }` will always put `value` before `other`. The default compare function is `{|a, b| a < b }`.